src/BottomNavigation/BottomNavigationButton.js

```diff
@@ -0,0 +1,117 @@
+// @flow weak
+
+import React, { Component, PropTypes, cloneElement, isValidElement } from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import classNames from 'classnames';
+import ButtonBase from '../internal/ButtonBase';
+
+export const styleSheet = createStyleSheet('BottomNavigationButton', (theme) => {
+  const { palette, typography, transitions } = theme;
+
+  return {
+    root: {
+      transition: `${transitions.create('color', '250ms')}, ${transitions.create('padding-top', '250ms')}`,
+      paddingTop: 8,
+      paddingBottom: 10,
+      paddingLeft: 12,
+      paddingRight: 12,
+      minWidth: 80,
+      maxWidth: 168,
+      background: 'none',
+      color: palette.text.secondary,
+      flex: '1',
+    },
+    selected: {
+      paddingTop: 6,
+      color: palette.primary[500],
+    },
+    selectedIconOnly: {
+      paddingTop: 16,
+    },
+    label: {
+      fontSize: 12,
+      opacity: 1,
+      transition: 'font-size 0.2s, opacity 0.2s',
+      transitionDelay: '0.1s',
+    },
+    selectedLabel: {
+      fontSize: typography.fontSize,
+    },
+    hiddenLabel: {
+      opacity: 0,
+      transitionDelay: '0s',
+    },
+    icon: {
+      display: 'block',
+    },
+  };
+});
+
+export default class BottomNavigationButton extends Component {
+  handleChangeIndex = () => {
+    const { onChangeIndex, index } = this.props;
```

Comment:
I think that we should handle the `onClick` property as we are overriding the behavior.